### PR TITLE
Add integration tests for firewall CRUD operations

### PR DESCRIPTION
**Pull request by bentranter**
Adds the integration tests for firewall CRUD operations. Next up are all the commands affecting a firewall's Droplets, tags, and inbound/outbound rules.

### CODE REVIEW

File: integration/firewall_get_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	it.Before(func() {
+		expect = require.New(t)
+
+		server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			switch req.URL.Path {
+			case "/v2/firewalls/e4b9c960-d385-4950-84f3-d102162e6be5":
+				auth := req.Header.Get("Authorization")
+				if auth != "Bearer some-magic-token" {
+					w.WriteHeader(http.StatusUnauthorized)
+					return
```
Comment (zachgersh) [inline integration/firewall_get_test.go]: Missing the method check?

File: integration/firewall_list_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		expect = require.New(t)
+
+		server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			switch req.URL.Path {
+			case "/v2/firewalls":
+				auth := req.Header.Get("Authorization")
+				if auth != "Bearer some-magic-token" {
+					w.WriteHeader(http.StatusUnauthorized)
+					return
+				}
```
Comment (zachgersh) [inline integration/firewall_list_test.go]: Also missing method check!

File: integration/firewall_list_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+				cmd := exec.Command(builtBinaryPath,
+					"-t", "some-magic-token",
+					"-u", server.URL,
+					"compute",
+					"firewall",
+					alias,
+				)
+
+				output, err := cmd.CombinedOutput()
+				expect.NoError(err, fmt.Sprintf("received error output from command %s: %s", alias, output))
```
Comment (zachgersh) [inline integration/firewall_list_test.go]: very nice on adding which command is failing!

File: integration/firewall_update_test.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	it.Before(func() {
+		expect = require.New(t)
+
+		server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			switch req.URL.Path {
+			case "/v2/firewalls/e4b9c960-d385-4950-84f3-d102162e6be5":
+				if req.Method != http.MethodPut {
+					w.WriteHeader(http.StatusMethodNotAllowed)
+					return
+				}
```
Comment (zachgersh) [inline integration/firewall_update_test.go]: nit: can we move this after the auth check? Matches the flow of the other tests.

Comment (bentranter) [reply integration/firewall_update_test.go]: Sure 👍

File: integration/firewall_get_test.go (hunk 1 above)
Comment (bentranter) [reply integration/firewall_get_test.go]: Whoops, I assumed we didn't bother for idempotent methods for some reason – I'll add them

**Pull request merged**